lib/api/controllers/documentController.js

```diff
@@ -146,9 +146,13 @@ class DocumentController extends BaseController {
    * @returns {Promise<Object>}
    */
   mGet (request) {
-    const
-      ids = this.getBodyArray(request, 'ids'),
-      { index, collection } = this.getIndexAndCollection(request);
+    let ids;
+    if (request.input.body) {
+      ids = this.getBodyArray(request, 'ids');
+    } else if (request.input.resource) {
+      ids = this.getIds(request);
+    }
+    const { index, collection } = this.getIndexAndCollection(request);
 
     this.assertNotExceedMaxFetch(ids.length);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') TypeError can be thrown as "ids" might be null or undefined here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2259)